schema/validator.go

```diff
@@ -86,13 +106,28 @@ func (v Validator) Validate(src io.Reader) error {
 	}
 }
 
-type unimplemented string
+// ValidateRefMedia validates the referenced objects against OCI media type defined in spec.
+// r: the given reader.
+// v: the expected media type.
+func ValidateRefMedia(r io.Reader, v Validator) error {
+	f, ok := vlidateRefMediaMap[v]
+	if ok {
+		if f == nil {
+			return fmt.Errorf("referenced media type validation %q unimplemented", v)
+		}
 
-func (v unimplemented) Validate(src io.Reader) error {
-	return fmt.Errorf("%s: unimplemented", v)
+		buf, err := ioutil.ReadAll(r)
+		if err != nil {
+			return errors.Wrap(err, "unable to read the document file")
+		}
+
+		return f(bytes.NewReader(buf))
+	}
+
+	return nil
```

Comment:
> Some validations such as descriptor, layout doesn't exist in function map, which make no sense.

We can validate descriptors against the JSON Schema.  And if you want JSON Schema validation handled elsewhere, we can define a no-op validator for types where we know there are no children.  But I think silently ignoring validating for all types that have no validator registered is setting ourselves up for “oops, that wasn't valid, but we forgot to add a validator for $TYPE”.